Doc/Manual/Warnings.html

```diff
@@ -535,6 +535,7 @@ <H3><a name="Warnings_nn14">18.9.5 Code generation (500-559)</a></H3>
 <li>522. Use of an illegal constructor name '<em>name</em>' in %extend is deprecated, the constructor name should be '<em>name</em>'.
 <li>523. Use of an illegal destructor name '<em>name</em>' in %extend is deprecated, the destructor name should be '<em>name</em>'.
 <li>524. Experimental target language. Target language <em>language</em> specified by <em>lang</em> is an experimental language. Please read about SWIG experimental languages, <em>htmllink</em>.
+<li>525. The director base class '<em>name</em>' or the destructor of director base class '<em>name</em>' is marked as final.
```

Comment:
The documentation of this warning is different to the actual warning issued. Perhaps you intended to issue this warning whenever a final method was parsed and then removed the warning?

It looks like a class with a final destructor and is marked with feature director will not be a director class and then you get the warning. I think the coding of the implementation in the patch is best wrt expected warnings and behaviour. I'd like to see a couple of lines of documentation in the Directors section stating what is probably pretty obvious. That is virtual final methods are ignored without warning. Any class marked with director issues the warning and is equivalent to marking it with nodirector.

Reply:
I had intended to use this warning when support for final classes is added to SWIG. Marking a class' dtor as final is effectively the same as marking the whole class as final.